module/zfs/zfs_ioctl.c

```diff
@@ -3258,22 +3311,30 @@ zfs_ioc_clone(const char *fsname, nvlist_t *innvl, nvlist_t *outnvl)
 {
 	int error = 0;
 	nvlist_t *nvprops = NULL;
+	nvlist_t *hidden_args = NULL;
+	dsl_crypto_params_t *dcp = NULL;
 	char *origin_name;
 
 	if (nvlist_lookup_string(innvl, "origin", &origin_name) != 0)
 		return (SET_ERROR(EINVAL));
 	(void) nvlist_lookup_nvlist(innvl, "props", &nvprops);
+	(void) nvlist_lookup_nvlist(innvl, ZPOOL_HIDDEN_ARGS, &hidden_args);
```

Comment:
Fair enough.  Although I was thinking - there's really no harm in not encrypting the blocks that are written when the objset is created, because the user hasn't provided any data yet.  It's just boilerplate.  Might still be best to do it the way you have, but something to keep in mind in case we run into any more problems in the creation code paths.